src/main/scala/mesosphere/marathon/core/launcher/impl/InstanceOpFactoryImpl.scala

```diff
@@ -183,7 +183,7 @@ class InstanceOpFactoryImpl(
 
       resourceMatchResponse match {
         case matches: ResourceMatchResponse.Match =>
-          val instanceOp = launchOnReservation(runSpec, offer, volumeMatch.instance, matches.resourceMatch, volumeMatch)
+          val instanceOp = launchOnReservation(volumeMatch.instance.runSpec, offer, volumeMatch.instance, matches.resourceMatch, volumeMatch)
```

Comment:
Shouldn't you use the `runSpec` extracted in https://github.com/mesosphere/marathon/pull/6664/files#diff-b201fe360b90772b45e15a0a9df02227R152

Reply:
Nope. It was a bug from my side.